### PR TITLE
Check for empty snapshot list

### LINKED ISSUE #713: Installed page shows empty page when update icon is selected
This is because the if statement only checks whether the snapshot is not null, so an empty list would be ignored here.

https://user-images.githubusercontent.com/73116038/211177321-a37e3ad7-1c52-47b7-bab5-0e5769b563a0.mp4

### PR DESCRIPTION
**Pull request by whiskeyPeak**
I was also getting some layout exceptions because of the Expanded widget so I removed all that logic. The page looks the exact same as before in all 3 layouts.

@Feichtmeier lmk if I'm missing something though!

https://user-images.githubusercontent.com/73116038/211177448-87cf6aed-623d-4957-9d71-892d50a35439.mp4

Fixes #713

### CODE REVIEW

**Comment by Feichtmeier** [comment]
I'm pretty sure we need the expanded here to let the grid always get the full height. I'll double check later

**Comment by Feichtmeier** [comment]
@whiskeyPeak there is indeed a bit inconsitency with expanded here and there, I think it would be more transparent for re-use if the expanded is around a widget and not inside. Maybe some artefacts from moving stuff around too much. Do you mind if I fix this in a different PR, you remove the expanded changes here and just focus on the no updates screen?
In general it is better to focus on one task in a PR - I am often lazy myself I know this feeling - but it is better to make another PR then :smile:

File: lib/app/updates/no_updates_page.dart [hunk 1]
```diff
@@ -6,14 +6,11 @@ import 'package:yaru_icons/yaru_icons.dart';
...
     Key? key,
-    this.expand = true,
   }) : super(key: key);
 
-  final bool expand;
-
   @override
   Widget build(BuildContext context) {
-    final page = Center(
+    return Center(
```
Comment (Feichtmeier) [inline lib/app/updates/no_updates_page.dart:13]: Please remove the changes for "expanded", I fix this everywhere in a different PR

**Review by Feichtmeier — CHANGES_REQUESTED** [review]

File: lib/app/updates/no_updates_page.dart (hunk 1 above)
Comment (whiskeyPeak) [reply lib/app/updates/no_updates_page.dart]: Done!

**Comment by whiskeyPeak** [comment]
After the latest rebase, i don't get the exception anymore but the message and icon aren't centred anymore.

Screenshot from 2023-01-08 16-10-24

**Comment by Feichtmeier** [comment]
@whiskeyPeak wrap it inside Expanded -> Center -> Singlechildscrollview

**Comment by whiskeyPeak** [comment]
I think it makes more sense to have the centre widget be part of the noUpdatesPage (we always want to the icon and message to be centred), have the singleChildScrollView be wrapped outside in cases where there's pixel overflow and just not use the expanded widget (i get a bunch of exceptions when it's wrapped around centre).

**Comment by Feichtmeier** [comment]
those pages are sometimes part of a column and sometimes not. Thus sometimes they are expanded and sometimes not. Those scroll windows are needed for those tiny winy window sizes, the center is needed AROUND the scrolling window though. So we keep the noupates/updating pages are minimal as possible and leave it to the outer pages to place them. Thank you

**Review by Feichtmeier — APPROVED** [review]

**Comment by whiskeyPeak** [comment]
> those pages are sometimes part of a column and sometimes not. Thus sometimes they are expanded and sometimes not. Those scroll windows are needed for those tiny winy window sizes, the center is needed AROUND the scrolling window though. So we keep the noupates/updating pages are minimal as possible and leave it to the outer pages to place them. Thank you

Yeah, I see what you mean. Done!

**Pull request merged**